stdlib/2.7/io.pyi

```diff
@@ -74,7 +74,7 @@ class StringIO(TextIO):
 
 class TextIOWrapper(TextIO):
     # write_through is undocumented but used by subprocess
-    def __init__(self, buffer: IO[str], encoding: unicode = ...,
+    def __init__(self, buffer: IO[bytes], encoding: unicode = ...,
```

Comment:
Hm, I've got a feeling that the buffer argument here is actually more restricted than IO[] -- it probably needs to be derived from BufferedIOBase.